dhcp-relay/dhcp_kern_xdp.c

```diff
@@ -438,57 +631,50 @@ int xdp_dhcp_relay(struct xdp_md *ctx) {
 
 			if (write_dhcp_option_255(ctx, option_offset)) {
 				bpf_printk("Could not write DHCP option 255 at offset %i", option_offset);
-				return XDP_ABORTED;
+				//return XDP_ABORTED;
+				break;
 			}
 
 			bpf_printk("Wrote DHCP option 255 at offset %i, returning XDP_PASS", option_offset);
 
 			break;
 		}
+
 		pos++;
 
+		/* Verifier check */
+		if (pos + 1 > data_end) {
+			break;
+		}
+
 		option_length = *pos;
 		option_offset += option_length + 2;
 
+		/* Verifier check */
 		if (pos + 1 > data_end) {
 			break;
 		}
 		pos++;
 
+		/* Verifier check */
 		if (pos + option_length > data_end) {
 			break;
 		}
+
+		/* Skip option value (go to next option) */
 		pos += option_length;
 
 	}
 
-	//return XDP_PASS;
-
-	/* Copy stored headers from buf to context */
-	/*if (xdp_store_bytes(ctx, 0, buf, static_offset, 0)) {
-
-		bpf_printk("xdp_store_bytes(ctx, 0, buf, %i) failed", static_offset);
-		return XDP_ABORTED;
-	}*/
-
-
-	/* make space for option 82 - copy DHCP options after increasing offset */
-	/*if (offset > static_offset) {
-		offset = static_offset;
-		for (i = 0; i < VLAN_MAX_DEPTH; i++) {
-			if (vlans.id[i]) {*/
-	/*  */
-	/*if (xdp_store_bytes(ctx, offset, buf + offset,
-		4, 0))
-		return XDP_ABORTED;
-	offset += 4;
-}
+	/* Adjust IP offset for VLAN header when VLAN header has been added */
+	if (head_adjusted) {
+		ip = data + ip_offset + sizeof (struct vlan_hdr);
```

Comment:
Why carry this `head_adjusted` all the way down here instead of just modifying `ip_offset` directly when doing the adjustment?